### PR TITLE
add a flag to wrap models/tests in jinja blocks

### LINKED ISSUE #1400: Wrap DBT models/tests in `{% model %}...{% endmodel %} blocks
## Feature
Wrap DBT models/tests in `{% model %}...{% endmodel %}` blocks

### Feature description
When we read in a sql file that is a model, we should optionally wrap it in a `{% model %}` block. Then we should run dbt's top-level jinja block parser over it and make sure the output matches the contents of the sql file, and either raise or log a warning.

Model blocks should probably be named using the basename of the sql file they were found in just for coherency, but it doesn't really matter.

This should be behind a (hidden?) flag to dbt compile/run.

We should do the same when generating tests - after we build the string, wrap it in a block, extract that block, and make sure it matches.

### Who will this benefit?
This will help us shake out any bugs around block parsing.

### PR DESCRIPTION
**Pull request by beckjake**
And then run all tests through that. Everything seems to have worked fine!

Fixes #1400

### CODE REVIEW

File: core/dbt/parser/base.py [hunk 1]
```diff
@@ -265,3 +265,43 @@ def parse_node(self, node, node_path, package_project_config, tags=None,
...
+                'from its wrapped block, expected "{}"'
+                .format(block.block_name, name, path, name)
+            )
+
+        if blocks[0].contents != contents:
+            raise dbt.exceptions.InternalException(
+                'Found invalid block contents when parsing node {} ({}) '
+                'from its wrapped block, expected it to match original file'
+                .format(name, path)
+            )
```
Comment (cmcarthur) [inline core/dbt/parser/base.py]: this is interesting, but it seems almost tautological.... unless someone included an unclosed comment or a `{% endmodel %}` in their model i would expect this to always pass. I am more interested in just running the toplevel block parser over the _unwrapped_ model contents, and comparing any compiler errors to those raised by the jinja parser. if the block parser raises an error, but the jinja parser does not, then we have a bug that we need to fix.

you should do the exact same thing for macro files too. `--wrap-models-in-tags` is probably a bad name if this turns the parser on for macros and models, so maybe `--test-new-parser` or something like that would be a better name

Comment (beckjake) [reply core/dbt/parser/base.py]: >unless someone included an unclosed comment or a `{% endmodel %}` in their model i would expect this to always pass

Well, me too, but I thought the idea was to shake out bugs around top-level block extraction that _are_ in that category. I guess I misunderstood - I'll change it to be the way you've described, which seems like a reasonable enough test.

Comment (cmcarthur) [reply core/dbt/parser/base.py]: @beckjake gotcha. I care more about comparing our parser to jinja's parser on arbitrary jinja code, not necessarily testing model block extraction. unrelated: i think wrapping model files in model blocks is a good way to achieve forwards compatibility with this new parser.

Comment (beckjake) [reply core/dbt/parser/base.py]: Yup! I that is the longer-term plan, I think. Maybe we can get it in for wilt-chamberlain if there isn't too much already, that would be nice.

**Pull request merged**